File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/TeleportProgress.java
package com.lumiyaviewer.lumiya.slproto.messages;

import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;
import java.util.UUID;

public class TeleportProgress extends SLMessage {
    public AgentData AgentData_Field = new AgentData();
    public Info Info_Field = new Info();

    public static class AgentData {
        public UUID AgentID;
    }

    public static class Info {
        public byte[] Message;
        public int TeleportFlags;
    }

    public TeleportProgress() {
        this.zeroCoded = false;
    }

    public int CalcPayloadSize() {
        return this.Info_Field.Message.length + 5 + 20;
    }

    public void Handle(SLMessageHandler sLMessageHandler) {
        sLMessageHandler.HandleTeleportProgress(this);
    }

    public void PackPayload(ByteBuffer byteBuffer) {
        byteBuffer.putShort(-1);
        byteBuffer.put((byte) 0);
        byteBuffer.put((byte) 66);
        packUUID(byteBuffer, this.AgentData_Field.AgentID);
        packInt(byteBuffer, this.Info_Field.TeleportFlags);
        packVariable(byteBuffer, this.Info_Field.Message, 1);
    }

    public void UnpackPayload(ByteBuffer byteBuffer) {
        this.AgentData_Field.AgentID = unpackUUID(byteBuffer);
        this.Info_Field.TeleportFlags = unpackInt(byteBuffer);
        this.Info_Field.Message = unpackVariable(byteBuffer, 1);
    }
}

File: Linkpoint/src/main/java/com/linkpoint/slproto/messages/TeleportProgress.kt
package com.linkpoint.slproto.messages

import com.linkpoint.slproto.SLMessage
import java.nio.ByteBuffer
import java.util.UUID

class TeleportProgress : SLMessage() {
    val AgentData_Field = AgentData()
    val Info_Field = Info()

    class AgentData {
        var AgentID: UUID? = null
    }

    class Info {
        var TeleportFlags: Int = 0
        lateinit var Message: ByteArray
    }

    init {
        zeroCoded = false
    }

    override fun CalcPayloadSize(): Int {
        return Info_Field.Message.size + 5 + 20
    }

    override fun Handle(handler: SLMessageHandler) {
        handler.HandleTeleportProgress(this)
    }

    override fun PackPayload(buffer: ByteBuffer) {
        buffer.putShort(-1)
        buffer.put(0.toByte())
        buffer.put(66.toByte())
        packUUID(buffer, AgentData_Field.AgentID)
        packInt(buffer, Info_Field.TeleportFlags)
        packVariable(buffer, Info_Field.Message, 1)
    }

    override fun UnpackPayload(buffer: ByteBuffer) {
        AgentData_Field.AgentID = unpackUUID(buffer)
        Info_Field.TeleportFlags = unpackInt(buffer)
        Info_Field.Message = unpackVariable(buffer, 1)
    }
}